compiler/rustc_mir_transform/src/check_null.rs

```diff
@@ -0,0 +1,73 @@
+use rustc_index::IndexVec;
+use rustc_middle::mir::interpret::Scalar;
+use rustc_middle::mir::*;
+use rustc_middle::ty::{Ty, TyCtxt};
+use rustc_session::Session;
+
+use crate::check_pointers::check_pointers;
+
+pub(super) struct CheckNull;
+
+impl<'tcx> crate::MirPass<'tcx> for CheckNull {
+    fn is_enabled(&self, sess: &Session) -> bool {
+        sess.ub_checks()
+    }
+
+    fn run_pass(&self, tcx: TyCtxt<'tcx>, body: &mut Body<'tcx>) {
+        check_pointers(tcx, body, &[], insert_null_check);
```

Comment:
Note that zero-sized reads/writes from null pointers are actually allowed. Does your pass handle that correctly?

Reply:
(This is also a missing test case)

Reply:
Very good point, I added that!